bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/DataPointFactory.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ism8.server;
+
+import java.util.ArrayList;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DataPointFactory} creates the data points depending on the types
+ *
+ * @author Hans-Reiner Hoffmann - Initial contribution
+ */
+@NonNullByDefault
+public class DataPointFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataPointFactory.class);
+    private static ArrayList<IDataPoint> DATA_POINTS = new ArrayList<>();
```

Comment:
The static method `createDataPoint` is creating the concrete DataPoints. Within this method I wanted to avoid that 2 instances are created pointing on the same ID. For this reason I needed a static list, holding a reference of all created DataPoints. In the server is a similar check and this takes care for creating only one DataPoint. After adapting the server (as suggested a bit later in this review) does make this obsolete. I'll remove this static list.